designs/2019-extensions-property-in-config/README.md

```diff
@@ -0,0 +1,63 @@
+- Start Date: 2019-11-12
+- RFC PR: https://github.com/eslint/rfcs/pull/48
+- Authors: Pig Fang ([@g-plane](https://github.com/g-plane)), Sindre Sorhus ([@sindresorhus](https://github.com/sindresorhus))
+
+# Specifying Extensions in Configuartion File
+
+## Summary
+
+This RFC proposes adding one property called `extensions` at top level of ESLint configuration files.
+
+## Motivation
+
+Currently if we want to specifying linting targets with specified extensions, the only way is to pass `--ext` via CLI, which isn't shareable. If specifying extensions in configuration file (`.eslintrc.*`) is supported, it will be convenient for shareable configurations files.
+
+## Detailed Design
+
+In this RFC, one property will be added to top level of configuration files:
+
+```json
+{
+    "extensions": ["ts", "vue"]
```

Comment:
given that unknown properties results in an error, wouldn't that mean that adding this config setting to a shared config is a breaking change?

imo, features for shared configs that are breaking changes are not truly for shared configs :-/